APE17.rst

```diff
@@ -0,0 +1,701 @@
+expected as long as it is included as a submodule. However, we recommend that
+astropy-helpers no longer be actively developed, in which case guarantees could
+not be made that some aspects of the astropy-helpers infrastructure will not
+break with future releases of Python, setuptools, or sphinx for example.
+
+However, we believe that the initial effort to switch over to the new guidelines
+will be a worthwhile investment - in all cases it will mean being able to get
+rid of astropy-helpers as a submodule and the confusion and headaches this can
+cause, and it will make for example the definition of package data much simpler
+and not have to worry about ``setup_package.py`` files in most cases.
+
+The astropy `package-template`_ will be updated to reflect the latest
+recommendations, which will make it easier for developers to update their
+packages.
+
+Impact for contributors
+-----------------------
+
+Users who wish to contribute fixes to the core astropy package or other packages
+will be encouraged to have `tox`_ installed if they want to easily run
+tests or build documentation locally. However, this is an easy package to
+install with `pip`_ and we could also add code in ``setup.py`` so that running
+``python setup.py test`` or ``python setup.py build_docs`` gives a helpful
+error message with instructions on updating and using tox to run tests and build
+the documentation.
+
+We note however that using tox is just a convenience and will not be compulsory,
+especially for pure-Python packages where running pytest directly will work.
+
+Impact on core astropy developers
+---------------------------------
+
+One of the main benefits of these changes will be to not have to maintain
+astropy-helpers any longer. Some parts of astropy-helpers have relied on hacks
+that can be brittle and break with new setuptools or Sphinx releases.
+Maintenance will still be needed for the proposed extension-helpers package but
+this will be a much smaller package than astropy-helpers, and by making it more
+generic and usable by any package, we hope to attract contributions from beyond
+the Astropy team.
+
+Impact on package managers
+--------------------------
+
+By relying on standard packaging infrastructure, this should facilitate the job
+of people involved in package managers - for example, the conda-forge
+infrastructure currently recommends that recipes should use pip to build
+packages, but this is not possible or easy at the moment for packages using
+astropy-helpers since they need to pass custom flags to setup.py to prevent the
+default behavior of checking for astropy-helpers releases online.
+
+Backward compatibility
+----------------------
+
+The changes described here are opt-in, so barring any breaking changes in
+Python, setuptools, or Sphinx, everything should continue to work as expected if
+developers do not make any changes.
+
+Alternatives
+------------
+
+In the words of Stuart: *Do nothing, suffer submodules and maintaining the
```

Comment:
I think it's worth stating that `pip` will probably eventually break the helpers (if that's true?).  That's a stronger "status quo won't work"/there are no alternatives argument.